packages/react/src/components/Popover/Popover.tsx

```diff
@@ -176,12 +176,43 @@ const arrowPseudoElement = {
   name: 'ArrowPseudoElement',
   fn(data: MiddlewareState) {
     const { elements, rects, placement } = data;
-    const arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
-    const arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+
+    let arrowX = 0;
+    let arrowY = 0;
+
+    if (placement.includes('bottom') || placement.includes('top')) {
+      /* default `arrowX` is if popover is bigger than the reference */
+      arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
+
+      arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+      /**
+       * if popover is wider, set arrow in the middle of the reference
+       */
+      if (rects.reference.width > rects.floating.width) {
+        arrowX = rects.floating.width / 2 - ARROW_HEIGHT / 2;
+      }
+    } else if (placement.includes('left') || placement.includes('right')) {
+      /* default `arrowX` is if popover is bigger than the reference */
+      arrowX = rects.reference.height / 2 + rects.reference.x - data.x;
```

Comment:
Is this supposed to be `height` or `width`?

Reply:
Is it working as intended by chance or just a coincident? This is how you get bugs 😅

Reply:
It is working as expected, I changed this around until I got expected behaviour. Will double check and add some comments

Reply:
updated comments in code, did some cleaning. Still looks good